frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-external/details-card.tsx

```diff
@@ -48,20 +55,29 @@ export const DetailsCard: React.FC<DashboardItemProps> = ({
     };
   }, [watchK8sResource, stopWatchK8sResource]);
 
-  const { ocs, subscription } = resources;
+  const { ocs, subscription, cephSecret } = resources;
   const ocsLoaded = ocs?.loaded || false;
   const ocsError = ocs?.loadError;
   const ocsData = ocs?.data;
+  const secretData = cephSecret?.data;
+  const secretLoaded = cephSecret?.loaded || false;
+  const secretError = cephSecret?.loadError;
   const ocsName = getName(ocsData?.[0]);
   const subscriptionLoaded = subscription?.loaded;
   const subscriptionError = subscription?.loadError;
   const subscriptionVersion = getOCSVersion(subscription as FirehoseResult);
 
-  const [infrastructure, infrastructureLoaded, infrastructureError] = useK8sGet<K8sResourceKind>(
-    InfrastructureModel,
-    'cluster',
-  );
-  const infrastructurePlatform = getInfrastructurePlatform(infrastructure);
+  const memoizedSecret: SecretKind = useDeepCompareMemoize(secretData, true) as any;
+  const link = React.useMemo(() => {
+    try {
+      const l = memoizedSecret.data.userKey;
+      const decode = Base64.decode(l);
+      return decode;
+    } catch {
+      return '';
+    }
+  }, [memoizedSecret]);
```

Comment:
Not using `useDeepCompareMemoize` is better, I agree...
One suggestion, it is better to use `useCallback` for `getCephLink` fucntion so that we do not Base64.decode the secret again if some other resource (other than secret) changes in the current FC... wdyt, @afreen23 ??

Reply:
Its okay until we see any performance issues.

Reply:
What I meant that, we are watching for 3 resources total in that particular FC... `OCSServiceModel`, `SubscriptionModel` & `SecretModel` ...
So, if `SecretModel` does not change, but, any of `OCSServiceModel`, `SubscriptionModel` changes then also it will decode the sercet again.

But, yeah this also make sense --> Its okay until we see any performance issues.